packages/interface/src/authority.ts

```diff
@@ -1,26 +1,21 @@
-import * as UCAN from '@ipld/dag-ucan'
-import type { DID, Identity, Signer, Verifier, Signature } from '@ipld/dag-ucan'
+export * as UCAN from '@ipld/dag-ucan'
+import type * as UCAN from '@ipld/dag-ucan'
 
-export type { DID, Identity, Signer, Verifier, Signature }
-
-export interface AuthorityParser<A extends number = number> {
-  parse(did: UCAN.DID): Authority<A>
-
-  // or<B extends number>(parser: AuthorityParser<B>): AuthorityParser<A | B>
+export interface PrincipalParser<A extends number = number> {
+  parse(did: UCAN.DID): Principal<A>
 }
 
-export interface Authority<A extends number = number>
+export interface Principal<A extends number = number>
   extends ArrayBufferView,
-    UCAN.Authority<A> {
+    UCAN.Verifier<A> {
   bytes: Uint8Array
 }
 
-export interface SigningAuthority<A extends number = number>
+export interface Agent<A extends number = number>
```

Comment:
Please don't get mad for going flip flopping on this, but I do have second thoughts on `Agent` which really should be what we call `Client` today.

That said I think I have good plan here which is we could instead do following:

```ts
export interface Verifier<A extends number = number>
  extends ArrayBufferView,
    UCAN.Verifier<A> {
  bytes: Uint8Array
}

export interface Signer<A extends number = number> extends ArrayBufferView,
    UCAN.Verifier<A>,
    UCAN.Signer<A> {
  verifier: Verifier<A>
  bytes: Uint8Array
  // secret: Uint8Array
}

export type Principal<A> =
  | Signer<A>
  | Verifier<A>
```

That way when we can refer to arbitrary principal as `Principal` but when we need to be specific we could say `Principal.Signer`.

Reply:
can we change it in a follow up ?